### PR TITLE
feat(datasets): add datasets endpoint

### PR DESCRIPTION
**Pull request by philloooo**
This implements the datasets endpoint that allows anonymous request(if configured) for tiered access feature.
ticket: https://ctds-planx.atlassian.net/browse/PXP-2463
deliverable: https://ctds-planx.atlassian.net/wiki/spaces/PLA/pages/29851651/Tiered+Access
cloud-automation pr: https://github.com/uc-cdis/cloud-automation/pull/699

### New Features
- Add /datasets endpoint to return summary counts for project and allow it to be configured to be public endpoint

### Deployment changes
- a new environment variable PUBLIC_DATASETS will be required to set for anonymous access to datasets endpoint

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
Jenkins Build 2 : time taken 1 min 38 sec
          Check the https://jenkins.planx-pla.net/job/perf-peregrine/job/PR-111/2/display/redirect

**Comment by PlanXCyborg** [comment]
Jenkins Build 3 : time taken 4 min 45 sec
          Check the https://jenkins.planx-pla.net/job/perf-peregrine/job/PR-111/3/display/redirect

**Comment by PlanXCyborg** [comment]
Jenkins Build 4 : time taken 19 min
          Check the https://jenkins.planx-pla.net/job/perf-peregrine/job/PR-111/4/display/redirect

File: peregrine/blueprints/datasets.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    if errors:
+        return flask.jsonify({"data": data, "errors": errors}), 400
+    result = {project_id: {} for project_id in projects}
+
+    for name, value in data.iteritems():
+        match = re.search("^i(\d)_(.*)", name)
+        index = int(match.group(1))
+        node = match.group(2)
+        result[projects[index]][node] = value
+    return flask.jsonify(result)
```
Comment (paulineribeyre) [inline peregrine/blueprints/datasets.py]: would it be safer to add `set_read_access_projects()` at the end?
right now `/graphql` does it anyway and it's the only other non-public endpoint, but what if we add another endpoint later

Comment (paulineribeyre) [reply peregrine/blueprints/datasets.py]: actually `set_read_access_projects()` does not do updates
https://github.com/uc-cdis/peregrine/blob/e189498b7fed9bfdc42a8807a1f1752cacd05897/peregrine/resources/submission/__init__.py#L87
so what happens if you request public datasets and then do a normal query? would you now have access to all projects?

File: peregrine/blueprints/datasets.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import flask
+import os
+import re
+
+import peregrine.utils
```
Comment (paulineribeyre) [inline peregrine/blueprints/datasets.py]: codacy is complaining about this not being used

File: peregrine/blueprints/datasets.py (hunk 1 above)
Comment (philloooo) [reply peregrine/blueprints/datasets.py]: flask.g is per request

**Comment by PlanXCyborg** [comment]
Jenkins Build 6 : time taken 1 min 21 sec
          Check the https://jenkins.planx-pla.net/job/perf-peregrine/job/PR-111/6/display/redirect

**Comment by PlanXCyborg** [comment]
Jenkins Build 7 : time taken 1 min 44 sec
          Check the https://jenkins.planx-pla.net/job/perf-peregrine/job/PR-111/7/display/redirect

**Comment by PlanXCyborg** [comment]
Jenkins Build 8 : time taken 1 min 55 sec
          Check the https://jenkins.planx-pla.net/job/perf-peregrine/job/PR-111/8/display/redirect

**Comment by PlanXCyborg** [comment]
Jenkins Build 9 : time taken 1 min 18 sec
          Check the https://jenkins.planx-pla.net/job/perf-peregrine/job/PR-111/9/display/redirect

**Pull request merged**